numba/unicode.py

```diff
@@ -1148,6 +1148,56 @@ def impl(a):
     return impl
 
 
+def _is_title(is_lower, is_upper, is_title):
+    # This algorithm is an amalgamation of:
+    # https://github.com/python/cpython/blob/master/Objects/unicodeobject.c#L11902-L11948
+    # https://github.com/python/cpython/blob/master/Objects/bytes_methods.c#L254-L291
+    def impl(a):
```

Comment:
I'm wondering whether all these `isFOO` methods could be generated? The code structure of the algorithm seems pretty much identical, so much that there's already an element of this here: https://github.com/numba/numba/pull/4442/files#diff-a21787256ad779b4c4c5c99915c71025R1181-R1185? Thoughts?